executor/explainfor_test.go

```diff
@@ -1392,7 +1392,7 @@ func (s *testPrepareSerialSuite) TestCTE4PlanCache(c *C) {
 	tk.MustExec("set @a=1, @b=2, @c=3, @d=4, @e=5, @f=0;")
 
 	tk.MustQuery("execute stmt using @f, @a, @f").Check(testkit.Rows("1"))
-	tk.MustQuery("execute stmt using @a, @b, @a").Check(testkit.Rows("1"))
+	tk.MustQuery("execute stmt using @a, @b, @a").Sort().Check(testkit.Rows("1", "2"))
 	tk.MustQuery("select @@last_plan_from_cache;").Check(testkit.Rows("0"))
```

Comment:
The correct result is '1 2': 

    drop table if exists t1;
    create table t1(a int);
    insert into t1 values(1);
    insert into t1 values(2);
    SELECT * FROM t1 dt WHERE EXISTS(WITH RECURSIVE qn AS (SELECT a*1 AS b UNION ALL SELECT b+2 FROM qn WHERE b=1) SELECT * FROM qn WHERE b=a);